go-controller/pkg/util/egressip_annotation.go

```diff
@@ -0,0 +1,69 @@
+package util
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+)
+
+const (
+	EgressIPMarkAnnotation = "k8s.ovn.org/egressip-mark"
+	EgressIPMarkBase       = 50000
+	EgressIPMarkMax        = math.MaxUint16
+)
+
+type EgressIPMark struct {
+	strValue string
+	intValue int
+}
+
+func (em EgressIPMark) String() string {
+	return em.strValue
+}
+
+func (em EgressIPMark) ToInt() int {
+	return em.intValue
+}
+
+func (em EgressIPMark) IsValid() bool {
+	return IsEgressIPMarkValid(em.intValue)
+}
+
+func (em EgressIPMark) IsAvailable() bool {
+	return em.strValue != ""
+}
+
```

Comment:
why is the add mark annotation not part of util and part of the patch logic in EIP controller?

Reply:
its shared logic so i stuffed it in util - I hate there is a util in ovn-k.. its an anti pattern.
EIP OVN controller handles the marks so i thought id make that uniform between CM and OVN EIP controller. I think EIP breth0 controller also needs to handle marks. I think wrapping the mark logic around this `EgressIPMark` struct makes it easier to ready the code.

Reply:
ack